source/base/function_signed_distance.cc

```diff
@@ -166,6 +166,42 @@ namespace Functions
 
 
 
+    template <int dim>
+    Tensor<1, dim>
+    Ellipsoid<dim>::gradient(const Point<dim> & point,
+                             const unsigned int component) const
+    {
+      AssertIndexRange(component, this->n_components);
+      (void)component;
+
+      const Point<dim> point_in_centered_cosy =
+        Point<dim>(compute_closest_point_ellipse(point) - center);
+
+      Tensor<1, dim> grad =
+        compute_analyical_normal_vector_on_ellipse(point_in_centered_cosy);
+      return grad / grad.norm();
```

Comment:
I don't understand this. Presumably, the function is of the form `(x/a)^2+(y/b)^2`. Then its gradient must vanish at the center of the ellipse, but you're making sure that the magnitude of the gradient is always one...

Reply:
![image](https://user-images.githubusercontent.com/44996996/161754233-af9043d0-4364-405b-b3e9-95b9d3c32ba9.png)
The signed distance function looks like this and is not `(x/a)^2+(y/b)^2 - 1 = 0`. I think the implemented gradient is correct.